subprojects/ide-native/src/main/java/org/gradle/ide/xcode/plugins/XcodePlugin.java

```diff
@@ -268,4 +297,85 @@ private static boolean isRoot(Project project) {
     private static String projectName(Project project) {
         return project.getName();
     }
+
+    private void registerXcodeProjectArtifact(Project project) {
+        ProjectLocalComponentProvider projectComponentProvider = ((ProjectInternal) project).getServices().get(ProjectLocalComponentProvider.class);
+        ProjectComponentIdentifier projectId = newProjectId(project);
+        projectComponentProvider.registerAdditionalArtifact(projectId, createXcodeProjectArtifact(projectId, xcodeModelFor(project), getProjectTask()));
+    }
+
+    private static LocalComponentArtifactMetadata createXcodeProjectArtifact(ProjectComponentIdentifier projectId, XcodeExtension xcode, Task projectTask) {
+        XcodeProject xcodeProject = xcode.getProject();
+        PublishArtifact publishArtifact = new XcodeProjectArtifact(xcodeProject, projectTask);
+        return new PublishArtifactLocalArtifactMetadata(projectId, publishArtifact);
+    }
+
+    private void addIncludedBuildToWorkspace(final Project project) {
+        if (isRoot(project)) {
+            final ServiceRegistry serviceRegistry = ((ProjectInternal) project).getServices();
+            getWorkspaceTask().dependsOn(new Callable<List<TaskDependency>>() {
```

Comment:
I would need more context on how to fix this as I don't think I know enough about the variant dependency management.